tools/net-common.in

```diff
@@ -63,6 +63,68 @@ function fix_mac {
   ip link set $INTERFACE address $FIXED_MAC || true
 }
 
+function bridge_vlan_add {
+  local VID="${1}"
+  shift
+
+  # when the VID is in use by traditional VLAN interface, then the
+  # tap on the VLAN aware bridge will not receive traffic
+  if [ -r /proc/net/vlan/config ]; then
+    if ! awk -F '[| ]*' -v vlan="^${VID}$" 'match($2, vlan) { exit 1 }' /proc/net/vlan/config; then
+      echo "VLAN ${VID} is in use by interface $(awk -F '[| ]*' -v vlan="^${VID}$" 'match($2, vlan) { print $1 }' /proc/net/vlan/config)"
```

Comment:
To avoid running awk twice here and make the code cleaner, you could store awk's result in a variable and check that directly:
```sh
if [ -r /proc/net/vlan/config ]; then
  vlan_interface="$(awk -F '[| ]*' -v vlan="^${VID}$" 'match($2, vlan) { print $1 }')"
  if [ -n "$vlan_interface" ]; then
   ....
  fi
fi